src/System.Net.HttpListener/src/System/Net/HttpListener.Windows.cs

```diff
@@ -0,0 +1,2135 @@
+                {
+                    throw new ArgumentException(SR.net_listener_slash, nameof(uriPrefix));
+                }
+                registeredPrefix = uriPrefix[j] == ':' ? String.Copy(uriPrefix) : uriPrefix.Substring(0, j) + (i == 7 ? ":80" : ":443") + uriPrefix.Substring(j);
+                fixed (char* pChar = registeredPrefix)
+                {
+                    i = 0;
+                    while (pChar[i] != ':')
+                    {
+                        pChar[i] = (char)CaseInsensitiveAsciiComparer.AsciiToLower[(byte)pChar[i]];
+                        i++;
+                    }
+                }
+                //GlobalLog.Print("HttpListener#" + LoggingHash.HashString(this) + "::AddPrefix() mapped uriPrefix:" + uriPrefix + " to registeredPrefix:" + registeredPrefix);
+                if (_state == State.Started)
+                {
+                    //GlobalLog.Print("HttpListener#" + LoggingHash.HashString(this) + "::AddPrefix() calling Interop.HttpApi.HttpAddUrl[ToUrlGroup]");
+                    uint statusCode = InternalAddPrefix(registeredPrefix);
+                    if (statusCode != Interop.HttpApi.ERROR_SUCCESS)
+                    {
+                        if (statusCode == Interop.HttpApi.ERROR_ALREADY_EXISTS)
+                            throw new HttpListenerException((int)statusCode, SR.Format(SR.net_listener_already, registeredPrefix));
+                        else
+                            throw new HttpListenerException((int)statusCode);
+                    }
+                }
+                m_UriPrefixes[uriPrefix] = registeredPrefix;
+                _defaultServiceNames.Add(uriPrefix);
+            }
+            //catch (Exception exception)
+            //{
+            //    if (NetEventSource.IsEnabled) NetEventSource.Exception(NetEventSource.ComponentType.HttpListener, this, "AddPrefix", exception);
+            //    throw;
+            //}
+            finally
+            {
+                //                if (NetEventSource.IsEnabled) NetEventSource.Exit(NetEventSource.ComponentType.HttpListener, this, "AddPrefix", "prefix:" + registeredPrefix);
+            }
+        }
+
+        internal bool ContainsPrefix(string uriPrefix) => m_UriPrefixes.Contains(uriPrefix);
+
+        public HttpListenerPrefixCollection Prefixes
+        {
+            get
+            {
+                //if (NetEventSource.IsEnabled) NetEventSource.Enter(NetEventSource.ComponentType.HttpListener, this, "Prefixes_get", "");
+                CheckDisposed();
+                if (_prefixes == null)
+                {
+                    _prefixes = new HttpListenerPrefixCollection(this);
+                }
+                return _prefixes;
+            }
+        }
+
+        internal bool RemovePrefix(string uriPrefix)
+        {
+            //if (NetEventSource.IsEnabled) NetEventSource.Enter(NetEventSource.ComponentType.HttpListener, this, "RemovePrefix", "uriPrefix:" + uriPrefix);
```

Comment:
All of this logging needs to be fixed/uncommented... will stop commenting on this.